### DISCUSSION TITLE
Navigate back in multiple layers (for example, close top layer)

### DISCUSSION THREAD

**Discussion by jogaman** [unpoly/unpoly#424] [Q&A] [answered]
Hi there,

I wonder how I go backwards in a existing hiearchy of overlays.

For example, if I click
`<a href="/layer1/" up-layer="new drawer">`
in that layer i click
`<a href="/layer1/layer2/" up-layer="new drawer">`
and in that layer i click
`<a href="/layer1/layer2/layer3/" up-layer="new drawer">`

I now have 3 overlays (drawers) open on top of each other. How do I go back to one specific level?

![alt text](https://i.imgur.com/7BBcB77.png)

Is there something like named overlays?

thanks!

**Comment by jogaman** [comment, marked as answer]
I did manage to solve it this way (instead of named overlays I identify them by the url)
```js

    up.link.config.followSelectors.push('a[href]')
    up.on('up:link:follow', function(event, link) {
        
        event.renderOptions.layer = 'new';
        event.renderOptions.position = 'right';
        event.renderOptions.mode = 'drawer';
        event.renderOptions.size = 'full';

        up.layer.overlays.forEach(overlay => {
            if(overlay.savedLocation === link.getAttribute('href')) {
                overlay.peel();
            }
        })
    })
```

It works, but I get an error in the console:

`Uncaught (in promise) up.Failed: Could not find common target in current page and response (tried selectors [[up-main~=drawer], [up-main~=overlay], [up-main=''], main] in new)`

**Reply by triskweline** [reply]
I always prefer to close overlays using [close conditions](https://unpoly.com/closing-overlays#close-condition), which can collapse an arbitrary number of layers.

I did not fully understand your solution, but if you're already working with `up:link:follow` you can set `event.renderOptions.layer` to an arbitrary `up.Layer` instance.


```js
up.on('up:link:follow', '.grandparent-layer', function(event, link) {
  event.renderOptions.layer = up.layer.get(up.layer.current.index - 2)
}
```

I made it so it only applies to links with the class `.grandparent-layer`, since we do not affect all the links on our page:

```html
<a href="/foo" class="grandparent-layer">...</a>
```

Note that using too many overlays will cause users to lose orientation in your app.

**Comment by jogaman** [comment]
Thank you!
This is great, now I can target specific overlays. I was looking for this in the docs but I still have to understand it better ;)
